### PR TITLE
feat(asset.party): add presence

### PR DESCRIPTION
**Pull request by DwifteJB**
## Description 

Adds presence to asset.party

## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

key
api-working

</details>

### CODE REVIEW

**Comment by DwifteJB** [comment]
new version of https://github.com/PreMiD/Presences/pull/7007

since the lint caused one of the checks to fail

File: websites/A/asset.party/metadata.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"id": "521762564062052393",
+		"name": "Dwifte#0661"
```
Comment (theusaf) [inline websites/A/asset.party/metadata.json:5]: Discriminator added incorrectly by old version of `pmd`
```suggestion
		"name": "Dwifte"
```

File: websites/A/asset.party/presence.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/* eslint-disable unicorn/prefer-switch */
```
Comment (theusaf) [inline websites/A/asset.party/presence.ts:1]: Why are we disabling eslint?
```suggestion
```

File: websites/A/asset.party/metadata.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	},
+	"service": "asset.party",
+	"description": {
+		"en": "Presence showing your activity on SBOX"
+	},
+	"url": "asset.party",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/vWSslVh.png",
```
Comment (theusaf) [inline websites/A/asset.party/metadata.json:13]: Logos should be 512x512px at minimum. Find or create a larger version (using something like waifu2x).

**Review by theusaf — CHANGES_REQUESTED** [review]

**Comment by theusaf** [comment]
> New version of #7007 Lint had caused the Presence Validator to fail.

You do not need to close and create a new PR. Simply push new commits to your branch, and it will automatically appear in this PR, and re-run the checks.

**Comment by DwifteJB** [comment]
I tried to for about 30 minutes and I was getting some stupid issues with git desktop, it also seemed I had a slightly outdated version of the repo which is why the other json files had linted

File: websites/A/asset.party/presence.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	// 	if (UsernameSelector[0] && UsernameSelector[0].innerHTML)
+	// 		Username = UsernameSelector[0].innerHTML;
+
+	// else
+	// 	console.log("failed!");
+	// }
+	if (pathname === "/") presenceData.details = "Exploring asset.party";
+	else if (pathname === "/get/developer/preview") {
+		const KeyDetails = document.querySelectorAll(".tag");
+		presenceData.details = "Going through key torture...";
```
[the remark this answers is not in the document]
Comment (DwifteJB) [reply websites/A/asset.party/presence.ts]: It's what the site called it, assumed it was safe for it to use. Changed anyway.

image

**Comment by DwifteJB** [comment]
Addressed all comments.

File: websites/A/asset.party/presence.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+		case "/api-history/": {
+			presenceData.details = "Viewing API History";
+			break;
+		}
+		default:
+			if (
+				document.querySelectorAll(".nav-thumb").length !== 0 &&
+				document.querySelectorAll(".page-header-block").length !== 0
+			) {
+				const Details = document.querySelectorAll(".page-header-block");
```
Comment (theusaf) [inline websites/A/asset.party/presence.ts:36]: `camelCase` should be used here too.

**Comment by DwifteJB** [comment]
Addressed camalCase comment.

File: websites/A/asset.party/presence.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
+const presence = new Presence({
+	clientId: "1042537481058394164",
+});
+
+enum Assets {
+	logo = "https://i.imgur.com/o5viesK.png",
+	smallKey = "https://i.imgur.com/mflNxaO.png",
+}
```
Comment (theusaf) [inline websites/A/asset.party/presence.ts:8]: Asset names should be capitalized.

**Review by theusaf — CHANGES_REQUESTED** [review]

**Comment by EncryptedDev** [comment]
Low quality

**Comment by DwifteJB** [comment]
gotta love when a few hours of your work just goes away after someone goes "low quality"

**Comment by EncryptedDev** [comment]
> gotta love when a few hours of your work just goes away after someone goes "low quality"

Take a look at the detail we expect from other presences. Try to include a range of features utilising timestamps, buttons, Records or what not. This is a fairly basic presence and isn't something we allow to be merged anymore.

You're very welcome to DM me on Discord (Encrypted#0001) via the community guild where I'll give you further information too

**Pull request closed without merging**